contracts/src/v0.8/automation/dev/chains/ArbitrumModule.sol

```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.19;
+
+import {ArbSys} from "./../../../vendor/@arbitrum/nitro-contracts/src/precompiles/ArbSys.sol";
+import {ArbGasInfo} from "./../../../vendor/@arbitrum/nitro-contracts/src/precompiles/ArbGasInfo.sol";
+import {IChainModule} from "../interfaces/v2_2/IChainModule.sol";
+
+contract ArbitrumModule is IChainModule {
+  /// @dev ARB_SYS_ADDR is the address of the ArbSys precompile on Arbitrum.
+  /// @dev reference: https://github.com/OffchainLabs/nitro/blob/v2.0.14/contracts/src/precompiles/ArbSys.sol#L10
+  address private constant ARB_SYS_ADDR = 0x0000000000000000000000000000000000000064;
+  ArbSys private constant ARB_SYS = ArbSys(ARB_SYS_ADDR);
+
+  /// @dev ARB_GAS_ADDR is the address of the ArbGasInfo precompile on Arbitrum.
+  /// @dev reference: https://github.com/OffchainLabs/nitro/blob/v2.0.14/contracts/src/precompiles/ArbGasInfo.sol#L10
+  address private constant ARB_GAS_ADDR = 0x000000000000000000000000000000000000006C;
+  ArbGasInfo private constant ARB_GAS = ArbGasInfo(ARB_GAS_ADDR);
+
+  function blockHash(uint256 n) external view returns (bytes32) {
+    uint256 blockNum = ARB_SYS.arbBlockNumber();
+    if (n >= blockNum || blockNum - n > 256) {
+      return "";
+    }
+    return ARB_SYS.arbBlockHash(n);
+  }
+
+  function blockNumber() external view returns (uint256) {
+    return ARB_SYS.arbBlockNumber();
+  }
+
+  function getL1Fee(bytes calldata) external view returns (uint256) {
+    return ARB_GAS.getCurrentTxL1GasFees();
+  }
```

Comment:
this function ignores the calldata and just uses the `msg.data`, right? This feels like a confusing API imho. Can we change this function to actually estimate the gas for the provided data?

Reply:
maybe we should change the interface to instead be `getCurrentL1Fee()` or similar and avoid passing data all together? Each module can read msg.data directly.